### ISSUE TITLE
lorax-composer.service stops responding to SIGTERM after query, times out on shutdown

### ISSUE THREAD

**Issue by martinpitt** [rhinstaller/lorax#283]
This is with the latest version 19.7.4-1.el7.centos from the COPR. Reproducer on current CentOS 7 image:
```
yum install -y epel-release
curl https://copr.fedorainfracloud.org/coprs/g/weldr/lorax-composer/repo/epel-7/group_weldr-lorax-composer-epel-7.repo > /etc/yum.repos.d/lorax-composer.repo
yum install -y lorax-composer 
systemctl start lorax-composer
curl http://localhost:4000/api/v0/projects/list
poweroff
```

The unit times out on shutdown:
```
[  OK  ] Started Show Plymouth Power Off Screen.
[  OK  ] Stopped Availability of block devices.
[...]
[ ***  ] A stop job is running for Lorax Ima...oser API Server (1min 27s / 1min 30s)
```
and eventually gets killed.

**Comment by martinpitt** [comment]
This only happens after doing an /api/v0/modules/list query. Not doing any query or just listing recipes does *not* cause the hang. I adjusted the reproducer in the description.

shutdown journal has no useful information from lorax itself, other than SIGTERM just timed out.

This actually reproduces pretty well without a shutdown:
```
# systemctl start lorax-composer
# systemctl stop lorax-composer  # fast
# systemctl start lorax-composer
# curl http://localhost:4000/api/v0/projects/list
# systemctl stop lorax-composer  # hangs; Control-C
# # systemctl status lorax-composer
● lorax-composer.service - Lorax Image Composer API Server
   Loaded: loaded (/usr/lib/systemd/system/lorax-composer.service; enabled; vendor preset: disabled)
   Active: deactivating (stop-sigterm) since Thu 2017-11-23 04:37:25 EST; 37s ago
```

So after processing a request it somehow stops responding to SIGTERM.

**Comment by martinpitt** [comment]
BTW, please avoid `Wants/After=network-online.target`. This is by and large a SysV legacy target which shouldn't be used any more, and there is little reason for a server like lorax-composer to do so. `network.target` is what you need here.

**Comment by bcl** [comment]
The SIGTERM problem is very likely yum mucking around with things. I noticed it when running things manually.

**Comment by bcl** [comment]
#287 should fix this

**Comment by martinpitt** [comment]
Confirmed fixed with 0.19.5, thanks!
